src/linter.cc

```diff
@@ -24,93 +24,235 @@
+absl::Status PrintASTTree(absl::string_view sql) {
     absl::Status return_status;
     std::unique_ptr<zetasql::ParserOutput> output;
 
-    return_status = zetasql::ParseStatement(sql,
-        zetasql::ParserOptions(), &output);
+    zetasql::ParseResumeLocation location =
+        zetasql::ParseResumeLocation::FromStringView(sql);
+    bool is_the_end = false;
+    int cnt = 0;
+    while ( !is_the_end ) {
+        return_status = zetasql::ParseNextScriptStatement(
+            &location, zetasql::ParserOptions(), &output, &is_the_end);
 
-    std::cout << "Status for sql \" " << sql << "\" = "
-        << return_status.ToString() << std::endl;
+        std::cout << "Status for sql#" << ++cnt << ": \"" << sql << "\" = "
+            << return_status.ToString() << std::endl;
 
-    if ( return_status.ok() ) {
-        std::cout << output -> statement() -> DebugString() << std::endl;
+        if ( return_status.ok() ) {
+            std::cout << output -> statement() -> DebugString() << std::endl;
+        } else {
+            break;
+        }
     }
-
     return return_status;
 }
 
-absl::Status checkStatement(absl::string_view sql) {
-    std::unique_ptr<zetasql::ParserOutput> output;
-    return zetasql::ParseStatement(sql,
-        zetasql::ParserOptions(), &output);
-}
-
-
-absl::Status checkLineLength(absl::string_view sql, int lineLimit,
+absl::Status CheckLineLength(absl::string_view sql, int line_limit,
     const char delimeter) {
     int lineSize = 0;
-    int lineNumber = 1;
-    for (int i=0; i<static_cast<int>(sql.size()); i++) {
+    int line_number = 1;
+    for (int i=0; i<static_cast<int>(sql.size()); ++i) {
         if ( sql[i] == delimeter ) {
             lineSize = 0;
-            lineNumber++;
+            ++line_number;
         } else {
-            lineSize++;
+            ++lineSize;
         }
-        if ( lineSize > lineLimit ) {
+        if ( lineSize > line_limit ) {
             return absl::Status(
                 absl::StatusCode::kFailedPrecondition,
-                absl::StrCat("Lines should be <= ", std::to_string(lineLimit),
-                " characters long [", std::to_string(lineNumber), ",1]") );
+                absl::StrCat("Lines should be <= ", std::to_string(line_limit),
+                " characters long [", std::to_string(line_number), ",1]") );
         }
     }
     return absl::OkStatus();
 }
 
-void KeywordExtractor::extract() {
-    // TODO(orhan_uysal): Implement a keyword extractor
-    // that pushes all keyword nodes to vector "keywords"
+absl::Status CheckStatement(absl::string_view sql) {
+    absl::Status return_status = absl::OkStatus();
+    std::unique_ptr<zetasql::ParserOutput> output;
+
+    zetasql::ParseResumeLocation location =
+        zetasql::ParseResumeLocation::FromStringView(sql);
+
+    bool is_the_end = false;
+    while ( !is_the_end && return_status.ok() ) {
+        return_status = zetasql::ParseNextScriptStatement(
```

Comment:
You can use RETURN_IF_ERROR macro here (https://github.com/google/zetasql/blob/master/zetasql/base/status_macros.h#L58 ):

```
while ( !is_the_end && return_status.ok() ) {
        RETURN_IF_ERROR(zetasql::ParseNextScriptStatement(...));
}
return absl::OkStatus();
```

The header, again, is not shared publicly, but we're using parser already anyways :D

Reply:
ok.